tools/lsp/preview/ui/property_view.rs

```diff
@@ -178,19 +180,166 @@ fn convert_number_literal(
+        }
+        expression_tree::Expression::Struct { values, .. } => {
+            println!("Stripping Struct");
+            if let Some(part) = palette_name.get(struct_level) {
+                eprintln!("  => field: {part}");
+                if let Some(value) = values.get(&SmolStr::from(part)) {
+                    eprintln!("    => with expr: {value:?}");
+                    return extract_color_from_evaluated_expression_impl(
+                        value,
+                        palette_name,
+                        struct_level.saturating_sub(1),
+                    );
+                }
+            }
+        }
+        expression_tree::Expression::Condition { true_expr, .. } => {
+            println!("Stripping Condition");
+            return extract_color_from_evaluated_expression_impl(
+                true_expr,
+                palette_name,
+                struct_level,
+            );
+        }
+        expression_tree::Expression::CodeBlock(cb) => {
+            println!("Stripping CodeBlock");
+            let Some(expr) = cb.first() else {
+                return ExtractedColor::default();
+            };
+            return extract_color_from_evaluated_expression_impl(expr, palette_name, struct_level);
+        }
+        expression_tree::Expression::StoreLocalVariable { value, .. } => {
+            println!("Stripping StoreLocalVariable");
+            return extract_color_from_evaluated_expression_impl(value, palette_name, struct_level);
+        }
+        _ => {
+            eprintln!("NOT HANDLED: {evaluated_expression:?}");
+        }
+    };
+    ExtractedColor::default()
+}
+
+fn extract_color_from_evaluated_expression(
+    evaluated_expression: &expression_tree::Expression,
+) -> ExtractedColor {
+    extract_color_from_evaluated_expression_impl(evaluated_expression, Vec::new(), 0)
+}
+
 fn extract_color(
-    expression: &syntax_nodes::Expression,
+    expression: &Option<syntax_nodes::Expression>,
+    evaluated_expression: &Option<expression_tree::Expression>,
     kind: ui::PropertyValueKind,
     value: &mut ui::PropertyValue,
 ) -> bool {
-    if let Some(text) = expression.child_text(SyntaxKind::ColorLiteral) {
-        if let Some(color) = ui::string_to_color(&text) {
-            value.display_string = text.as_str().into();
-            value.kind = kind;
-            value.value_brush = slint::Brush::SolidColor(color);
-            value.gradient_stops =
-                Rc::new(VecModel::from(vec![ui::GradientStop { color, position: 0.5 }])).into();
-            return true;
+    if let Some(expression) = expression {
+        if expression.children_with_tokens().any(|child| {
+            eprintln!("CODE TEST: Child kind: {:?}", child.kind());
+            ![SyntaxKind::QualifiedName, SyntaxKind::ColorLiteral].contains(&child.kind())
+        }) {
+            eprintln!("CODE IS TOO COMPLEX TO BE CONSIDERED");
```

Comment:
why do you need this test? you can let the extract_color_from_evaluated_expression decide of that.